src/util/mpir_pmi.c

```diff
@@ -642,11 +642,21 @@ int MPIR_pmi_allgather(const void *sendbuf, int sendsize, void *recvbuf, int rec
 /* This version assumes shm_buf is shared across local procs. Each process
  * participate in the gather part by distributing the task over local procs.
  *
+ * sendbufs     - An array of pointers to buffers to be contibuted.
+ * sendsizes    - An array of the sizes of the corresponding sendbufs.
+ * num_sendbufs - The number of buffers contributed by each process. Each process must contribute
+ *                the same number of buffers.
+ * shm_buf      - The output buffer where the results of the allgather will be stored.
+ * recvsize     - The maximum size of each contribution to the allgather. This value is used to
+ *              - calculate the offset of each item in the shm_buf buffer.
+ * domain       - The domain used to determine how many processes are contributing to the allgather.
+ *                Correct values are MPIR_PMI_DOMAIN_NODE_ROOTS and MPIR_PMI_DOMAIN_ALL.
+ *
  * NOTE: the behavior is different with MPIR_pmi_allgather when domain is
  * MPIR_PMI_DOMAIN_NODE_ROOTS. With MPIR_pmi_allgather, only the root_nodes participate.
  */
-int MPIR_pmi_allgather_shm(const void *sendbuf, int sendsize, void *shm_buf, int recvsize,
-                           MPIR_PMI_DOMAIN domain)
+int MPIR_pmi_allgatherv_shm(const void *sendbufs[], int sendsizes[], int num_sendbufs,
+                            void *shm_buf, int recvsize, MPIR_PMI_DOMAIN domain)
```

Comment:
It sounds like your concern is more about the name than the semantics. I would think that it's better to have all of the implementation of the PMI call in the same function instead of putting them in the caller. Both options are still available to the "user", but internally, they use the same code base, which simplifies implementation.

Reply:
I'm going to "resolve" this comment and we can continue below.